doc/how_to/custom_components/reactive_esm/reactive_esm_layout.md

```diff
@@ -0,0 +1,143 @@
+# Create Layouts With ReactiveESM
+
+In this guide we will show you how to build custom layouts using HTML and `ReactiveESM`.
+
+## Layout a single Panel Component
+
+You can layout a single Panel component as follows.
+
+:::::{tab-set}
+
+::::{tab-item} JavaScript
+
+```{pyodide}
+import param
+import panel as pn
+
+pn.extension()
+
+class LayoutSingleObject(pn.ReactiveESM):
+    object = param.ClassSelector(class_=pn.viewable.Viewable, allow_refs=False)
```

Comment:
Normal layouts like `pn.Column` would also accept non-`Viewable` objects and just apply `pn.panel` automatically. How do we support that?

Reply:
I'm planning to define parameter types for Viewable that converts on set (which will also be used by other layouts).